### ISSUE TITLE
Issue in installing Box.V2.JWTAuth package through Nuget

### ISSUE THREAD

**Issue by Soundaram** [box/box-windows-sdk-v2#144]
We are attempting to build an asp.net c# web application using c# code that connects & authenticates the users to our Box.Com account and displays file names according to the search performed using the Box Windows SDK V2

We have downloaded the v2.7.0 version from the url “https://github.com/box/box-windows-sdk-v2/releases”

Successfully installed Box.V2 package.

Unfortunately we are not able to add Box.V2.JWTAuth to our project. It's returning this error after trying to add it from nuget:

PM> Install-Package Box.V2.JWTAuth
Attempting to resolve dependency 'BouncyCastle (≥ 1.7.0)'.
Attempting to resolve dependency 'Microsoft.Net.Http (≥ 2.2.29)'.
Attempting to resolve dependency 'Microsoft.Bcl (≥ 1.1.10)'.
Attempting to resolve dependency 'Microsoft.Bcl.Build (≥ 1.0.14)'.
Attempting to resolve dependency 'System.IdentityModel.Tokens.Jwt (≥ 4.0.2.206221351)'.
Installing 'Box.V2.JWTAuth 1.0.3'.
Successfully installed 'Box.V2.JWTAuth 1.0.3'.
Adding 'BouncyCastle 1.7.0' to Box.V2.
Install failed. Rolling back...
Install-Package : Could not install package 'BouncyCastle 1.7.0'. You are trying to install this package into a project that 
targets 'portable-win+net40+sl50+wp80', but the package does not contain any assembly references or content files that are 
compatible with that framework. For more information, contact the package author.
At line:1 char:1
- Install-Package Box.V2.JWTAuth
- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  - CategoryInfo          : NotSpecified: (:) [Install-Package], InvalidOperationException
  - FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PowerShell.Commands.InstallPackageCommand

which in turn causes build errors also.

**Comment by cburnette** [comment]
Hi, the project that you are trying to add Box.V2.JWTAuth to must be a .Net 4.5 project and not a portable library project. 

Chad

**Comment by Soundaram** [comment]
Hi,
We were using .Net 4.5 framework. Also we tried with earlier versions of the SDK and ended up with same error.

Visual Studio Product Details of our Development machine:
1..Net Framework 4.5
2.Microsoft Visual Studio Premium 2012
Version 11.0.61219.00 Update 5
Microsoft .NET Framework
Version 4.5.50709
Installed Version: Premium
3.Visual C# 2012   04941-004-0043007-02573. Microsoft Visual C# 2012
4.NuGet Package Manager   2.8.60318.667. NuGet Package Manager in Visual Studio. 

Below are the attempt that we have tried:

Attempt 1:

1)	Downloaded the v2.7.0  version from the url “https://github.com/box/box-windows-sdk-v2/releases”
2)	Opened the Solution in the Visual Studio
3)	As suggested in the GitHub block, tried to execute the below commands in the Package Manager Console,

PM> Install-Package Box.V2
                                We are able to install the Box.V2 successfully.

           PM> Install-Package Box.V2.JWTAuth
              We ended up with the below error message:

Attempting to resolve dependency 'BouncyCastle (≥ 1.7.0)'.
Attempting to resolve dependency 'Microsoft.Net.Http (≥ 2.2.29)'.
Attempting to resolve dependency 'Microsoft.Bcl (≥ 1.1.10)'.
Attempting to resolve dependency 'Microsoft.Bcl.Build (≥ 1.0.14)'.
Attempting to resolve dependency 'System.IdentityModel.Tokens.Jwt (≥ 4.0.2.206221351)'.
Installing 'Box.V2.JWTAuth 1.0.3'.
Successfully installed 'Box.V2.JWTAuth 1.0.3'.
Adding 'BouncyCastle 1.7.0' to Box.V2.
Install failed. Rolling back...
Install-Package : Could not install package 'BouncyCastle 1.7.0'. You are trying to install this package into a project that targets 
'portable-win+net40+sl50+wp80', but the package does not contain any assembly references or content files that are compatible with 
that framework. For more information, contact the package author.
At line:1 char:1
+ Install-Package Box.V2.JWTAuth
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], InvalidOperationException
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PowerShell.Commands.InstallPackageCommand

4)	Even then, we tried to build the solution which thrown multiple errors, so we could not proceed further.

Attempt 2:

1)	Downloaded the v2.7.0  version from the url https://github.com/box/box-windows-sdk-v2/releases
2)	Opened the Solution in the Visual Studio
3)	This time, we didn’t install the packages “Box.V2” and “Box.V2.JWTAuth” and built the solution in Visual Studio
4)	In this case, we are able to build the package successfully without any errors
5)	Now,  as per the blogs and Box Documentation, we have two options to proceed with respect to authentication as listed below:
a.	OAuth2
b.	JWTAuth

OAuth2:

Navigated to “Box.V2.Samples.WPF” in the Visual Studio->ViewModels->MainViewModel.cs and have filled the ClientId , ClientSecret, Redirect UI as per our sandbox app.
And tried to debug the “Box.V2.Samples.WPF” solution where, 
-	We see a browser window getting opened and asking us the credentials to pass. 
We gave the admin account credentials , and able to see “Grant Access” option display in the browser window.
-	On clicking the Grant Access, as per the code logic we should get the authorization code, but we are not getting the authorization code.

Config = new BoxConfig(ClientId, ClientSecret, RedirectUri);

JWTAuth:

Navigated to “Box.V2.Samples.JWTAuth” in the visual studio->Program.cs

To work on this, we should install the “Box.V2.JWTAuth” package, but we are getting the error message while trying to install the package through package manager console. 
Have tried a lot of option like changing the target framework version, and installing the lower version of box SDK and etc.,. but we are not able to fix it.
